presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionTask.java

```diff
@@ -15,259 +15,196 @@
-                FutureCallback callback = new SimpleHttpResponseHandler<>(new TaskInfoHandler(), request.getUri(), stats.getHttpResponseStats(), NATIVE_EXECUTION_TASK_ERROR);
-
-                Futures.addCallback(
-                        future,
-                        callback,
-                        executor);
-            }
-            catch (Throwable t) {
-                throw t;
-            }
-        }, 0, GET_TASK_INFO_INTERVALS, MILLISECONDS);
+        CompletableFuture<Void> future = new CompletableFuture<>();
+        Futures.addCallback(
+                workerClient.updateTask(
+                        sources,
+                        planFragment,
+                        tableWriteInfo,
+                        session,
+                        outputBuffers),
+                new UpdateResponseHandler(future),
+                executor);
+        return future;
     }
 
     private class UpdateResponseHandler
-            implements SimpleHttpResponseCallback<TaskInfo>
+            implements FutureCallback<BaseResponse<TaskInfo>>
     {
-        @Override
-        public void success(TaskInfo value)
-        {
-            try (SetThreadName ignored = new SetThreadName("UpdateResponseHandler-%s", taskId)) {
-                log.debug("success " + value.getTaskId());
-                taskInfo.set(value);
-            }
-        }
+        private final CompletableFuture<Void> future;
 
-        @Override
-        public void failed(Throwable cause)
+        public UpdateResponseHandler(CompletableFuture<Void> future)
         {
-            try (SetThreadName ignored = new SetThreadName("UpdateResponseHandler-%s", taskId)) {
-                log.error("failed " + cause);
-            }
+            this.future = requireNonNull(future, "future is null");
         }
 
         @Override
-        public void fatal(Throwable cause)
+        public void onSuccess(BaseResponse<TaskInfo> result)
         {
             try (SetThreadName ignored = new SetThreadName("UpdateResponseHandler-%s", taskId)) {
-                log.error("fatal " + cause);
-            }
-        }
-    }
-
-    private class TaskInfoHandler
-            implements SimpleHttpResponseCallback<TaskInfo>
-    {
-        @Override
-        public void success(TaskInfo value)
-        {
-            try (SetThreadName ignored = new SetThreadName("TaskInfoHandler-%s", taskId)) {
-                log.debug("TaskInfoHandler success " + value.getTaskId());
-                taskInfo.set(value);
-            }
-        }
-
-        @Override
-        public void failed(Throwable cause)
-        {
-            try (SetThreadName ignored = new SetThreadName("TaskInfoHandler-%s", taskId)) {
-                log.error("TaskInfoHandler failed " + cause);
+                TaskInfo value = result.getValue();
+                log.debug("success " + value.getTaskId());
```

Comment:
It's always good to use `format("success %s", value.getTaskId())` for string concatenation. Same for everything else in the PR